rust-runtime/smithy-eventstream/src/buf/count.rs

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! A [`Buf`] implementation that counts bytes read.
+
+use bytes::Buf;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+/// A [`Buf`] implementation that counts bytes read.
+pub struct CountBuf<'a> {
+    buffer: &'a mut dyn Buf,
+    count: AtomicUsize,
```

Comment:
Originally, I was increasing the count in `chunk`, which meant I needed interior mutability since `chunk` takes `&self`. But then I realized that was a bug since `chunk` can be called multiple times before advancing, so I fixed that and forgot to reduce the AtomicUsize back down. Will fix.